lib/chat_app_web/live/single_room_live.ex

```diff
@@ -0,0 +1,78 @@
+defmodule ChatAppWeb.SingleRoomLive do
+  use ChatAppWeb, :live_view
+  alias Phoenix.PubSub
+  alias ChatApp.LiveviewMonitor
+  require Logger
+
+  @impl true
+  def mount(%{"id" => room_id}, _session, socket) do
+    topic = "room:" <> room_id
+    cond do
+      connected?(socket) ->
+        Logger.info("Liveview's second mount() call")
+        LiveviewMonitor.monitor(self(), __MODULE__, %{id: socket.id, room_id: room_id})
+        PubSub.subscribe(ChatApp.PubSub, topic)
+        broadcasted_message = {:update_messages, %{text: "anon joined the chat.", datetime_as_string: ""}}
+        PubSub.broadcast(ChatApp.PubSub, topic, broadcasted_message)
+      true -> Logger.info("Liveview's first mount() call")
+    end
+
+    [{room_id, room}] = :ets.lookup(:rooms, room_id)
+    new_socket =
+      socket
+      |> assign(room_id: room_id)
+      |> assign(room: room)
+      |> assign(topic: topic)
+      |> assign(messages: [])
+      |> assign(users: room.current_participants)
```

Comment:
Agreed, in fact the only reason why I wrote the assigns in that manner is because I thought I would get rid of a cryptic Elixir Dialyzer warning that was placed on that piece of code, but that didn't help.
The previous line of code had no map syntax and looked something like this:

```elixir
assign(socket, room_id: room_id, room: room, topic: topic)
```